### ISSUE TITLE
[Accessibility] Narrator doesn't announce number before Hyphen/Colon while deleting number in MaskedTextBox

### ISSUE THREAD

**Issue by Eudora-Li01** [dotnet/winforms#9391]
### .NET version

.NET 8.0.100-preview.7.23328.8 

### Did it work in .NET Framework?

No

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

No

Narrator doesn't announce all of numbers in MaskedTextBox while deleting numbers in .NET 6.0, 7.0 and .NET Framework.

### Issue description

1. Narrator doesn't announce numbers before Hypen/Colon while deleting numbers in MaskedTextBox

https://github.com/dotnet/winforms/assets/130345015/dec9afe2-1dec-4659-a4b4-debebf882ea5

2. It works well in NVDA for NET 6.0, 7.0, 8.0 and Framework. NVDA announce all of numbers while deleting numbers in MaskedTextBox.

### Steps to reproduce

1. Start a Winforms .Net Core application with MaskedTextBox
2. Set Mask as Data format with Hyphen/Colon (e.g MobilePhoneNumber)
3. Run app (F5)
4. Open Narrator
5. Input 12 digits number in MaskedTextBox
6. Delete number

**Comment by Tanya-Solyanik** [comment]
@Eudora-Li01 - Could you please investigate if WPF or MFC has similar control and how it behaves in this scenario?

**Comment by Eudora-Li01** [comment]
@Tanya-Solyanik , I verified on VS 17.8.0 Preview 1[33906.464] + .NET SDK 8.0.100-preview.7.23361.25, MFC have similar control MFC MaskedEdit Control. Narrator didn't announce the number while deleting them from last one.

https://github.com/dotnet/winforms/assets/130345015/12e121d2-b09e-4721-a8cd-afeca173b784

**Comment by Tanya-Solyanik** [comment]
Thank you @Eudora-Li01! Since winforms is consistent with MFC, we can close it

**Comment by merriemcgaw** [comment]
This is Common Controls, there isn't anything we should be doing in this case.
